scanners/trivy/trivy.py

```diff
@@ -0,0 +1,148 @@
+        # List important locations for host <-> container mapping point
+        # + work: where data is stored:
+        #   - Trivy config file, reports, evidence, etc. are beneath this path
+        self.path_map = None  # to be defined by the typed scanner
+
+    ###############################################################
+    # PUBLIC METHODS                                              #
+    # Called via inheritence only                                 #
+    ###############################################################
+
+    def setup(self):
+        """
+        This code handles only the "Trivy" layer, independently of the container used.
+        This method should not be called directly, but only via super() from a child's setup()
+        """
+        logging.info("Preparing Trivy configuration")
+        self._setup_trivy_cli()
+        self._setup_report()
+
+    def run(self):
+        """This code handles only the "Trivy" layer, independently of the container used.
+        This method should not be called directly, but only via super() from a child's setup()
+        This method is currently empty as running entirely depends on the containment
+        """
+        pass
+
+    def postprocess(self):
+        logging.info(f"Extracting report, storing in {self.results_dir}")
+        reports_dir = os.path.join(
+            self.path_map.workdir.host_path, Trivy.TMP_REPORTS_SUBDIR
+        )
+        shutil.copytree(reports_dir, self.results_dir)
+
+    def cleanup(self):
+        """Generic trivy cleanup: should be called only via super() inheritance"""
+
+        logging.debug(f"Deleting temp directory used {self.path_map.workdir.host_path}")
+        shutil.rmtree(self.path_map.workdir.host_path)
+        pass
+
+    ###############################################################
+    # PROTECTED METHODS                                           #
+    # Called via trivy or inheritence only                          #
+    # May be overloaded by inheriting classes                     #
+    ###############################################################
+
+    def _setup_trivy_cli(self):
+        """prepare the trivy command: self.trivy_cli
+        This is a list of strings, representing the entire Trivy command to match the desired run
+        """
+
+        # common options for both image and k8s(TBD)
+        cli_ops = []
+
+        # severity
+        severity = self.config.get("scanners.trivy.report.severity")
+        if severity:
+            cli_ops.extend(["--severity", severity])
+
+        # skip-db-update
+        if self.config.get("scanners.trivy.miscOptions.skipDbUpdate"):
+            cli_ops.extend(["--skip-db-update"])
+
+        image_conf = self.config.get("scanners.trivy.image")
+        if image_conf:
+            if self.config.get("general.container.type") == "podman":
+                # podman run <trivy_image> doesn't require 'trivy' command.
+                self.trivy_cli = ["image"]
```

Comment:
Ideally, this should be moved to trivy_podman somehow. 
Either in `_setup_podman_cli` or maybe using similar trick as you did in the __init__() with `Setup defaults specific to Podman`, for example storing that into `scanners.trivy.container.parameters.executable` [we already used this entry for ZAP, except that here we will need a list since it can contain 2 elements] => and in this _setup_trivy_cli(), simply `get()` it back

Reply:
good point. removed this. instead the trivy_cli is initiated in __init__() according to container.paramenters.executable